internal/pikamon/config.go

```diff
@@ -0,0 +1,47 @@
+package pikamon
+
+import (
+	"github.com/pkg/errors"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+// Config contains the various values that are configurable for the bot.
+type Config struct {
+	Discord struct {
+		Token string
+	}
+	Logging struct {
+		Level string
+	}
+}
+
+// GetConfig reads the config file and flags, then applies environment variable overrides.
+func GetConfig() (*Config, error) {
+	cfg := &Config{}
```

Comment:
So this is more of a learning question....I am reading this page:
- https://yourbasic.org/golang/structs-explained/

My question is this: Why use `&Config{}` instead of `Config{}`? From reading that page it seems like `Config{}` would work when making a struct. It has been a while since I had to mess with pointers, but I believe (correct me if I'm wrong) that this is giving you the address of a initialized Config struct? Which is odd because then you use below as
```
cfg.Logging.Level = viper.GetString("pikamon.logging.level")
```

Reply:
In Go, you generally want to pass pointers around except in specific cases. not only are they more efficient, but they help denote whether data is mutable or not. In the case of a config, you don't want to pass around all that data on every function call, passing the address is enough.

On the final point, pointers are generally how you pass around mutable data. So if you want to alter fields in a struct after creation, a pointer is safest and cheapest option.